extensions/amp-date-display/1.0/amp-date-display.js

```diff
@@ -76,14 +45,50 @@ class AmpDateDisplay extends PreactBaseElement {
     );
     return isLayoutSizeDefined(layout);
   }
+
+  /** @override */
+  checkPropsPostMutations() {
+    const templates =
+      this.templates_ || (this.templates_ = Services.templatesFor(this.win));
+    const template = templates.maybeFindTemplate(this.element);
+    if (template != this.template_) {
+      this.template_ = template;
+      if (template) {
+        // Only overwrite `render` when template is ready to minimize FOUC.
+        templates.whenReady(template).then(() => {
+          if (template != this.template_) {
+            // A new template has been set while the old one was initializing.
+            return;
+          }
+          this.mutateProps(
+            dict({
+              'render': (data) => {
+                return templates
+                  .renderTemplateAsString(dev().assertElement(template), data)
+                  .then((html) => dict({'__html': html}));
+              },
+            })
+          );
+        });
+      } else {
+        this.mutateProps(dict({'render': null}));
+      }
+    }
+  }
+
+  /** @override */
+  isReady(props) {
```

Comment:
Can you explain why this is needed?

Reply:
The idea is this: let's say we allow a placeholder markup like this:

```
<amp-date-display>
  <template>...</template>
  <time>will be rendered at some point</time>
</amp-date-display>
```

Since templates are rendered asynchronously and the time from start->render is undefined, we will see the following sequence:
1. `<time>will be rendered at some point</time>` - starting state.
2. `<time></time>` - first `DateDisplay` render with "no value" while `temlates.renderAsString()` is loading the template.
3. `<time>{actual value}</time>` - the `templates.renderAsString()` completes and `DateDisplay` is rerendered.

What `templates.whenReady()` and this `isReady()` do - they try to eliminate or minimize the (2) step, so that empty content doesn't flash and as much as possible we go from "will be rendered at some point" to "{actual value}".

From any other point of view this actually also helpful, I think. There's no much sense getting Preact executing and spending cycles before we know we are actually ready to do the work.

WDYT?

Reply:
Ohh, ok!